src/chat.cpp

```diff
@@ -1131,6 +1225,152 @@ CCMD( say_team )
 	}
 }
 
+//*****************************************************************************
+//
+// [AK] Allows players (or the server) to send private messages to other players.
+//
+bool chat_CanSendPrivateMessage( ULONG ulPlayer )
+{
+	if ( ulPlayer == MAXPLAYERS )
+		return true;
+
+	if ( g_ulChatPlayer == MAXPLAYERS )
+		g_ulChatPlayer = 0;
+
+	// [AK] If we're still trying to send a message to an invalid player, find another player.
+	if (( PLAYER_IsValidPlayer( g_ulChatPlayer ) == false ) || ( g_ulChatPlayer == (ULONG)consoleplayer ))
+	{
+		ULONG oldPlayer = g_ulChatPlayer;
+
+		do
+		{
+			// [AK] Keep looping until we find another player who we can send a message to.
+			if ( ++g_ulChatPlayer >= MAXPLAYERS )
+				g_ulChatPlayer = 0;
+
+			// [AK] If we're back to the old value, then there's nobody else to send a message to.
+			if ( g_ulChatPlayer == oldPlayer )
+			{
+				Printf( "There's no other player to send a private message to.\n" );
+				return false;
+			}
+		}
+		while (( PLAYER_IsValidPlayer( g_ulChatPlayer ) == false ) || ( g_ulChatPlayer == (ULONG)consoleplayer ));
```

Comment:
Looks like this will end in an infinite loop of there is only one player on the server.